worker/worker.go

```diff
@@ -64,8 +74,11 @@ func (w *Pool) StartWorkers() {
 					metrics.Timing("worker_processing_duration_milliseconds", (now.Sub(batchReadTime).Milliseconds())/lenBatch, "worker="+workerName)
 					metrics.Timing("server_processing_latency_milliseconds", (now.Sub(request.TimeConsumed)).Milliseconds()/lenBatch, fmt.Sprintf("conn_group=%s", request.ConnectionIdentifier.Group))
 				}
-				metrics.Count("kafka_messages_delivered_total", totalErr, fmt.Sprintf("success=false,conn_group=%s", request.ConnectionIdentifier.Group))
-				metrics.Count("kafka_messages_delivered_total", len(request.GetEvents())-totalErr, fmt.Sprintf("success=true,conn_group=%s", request.ConnectionIdentifier.Group))
+				for eventType, count := range producerStats.EventCounts {
```

Comment:
These metrics could be reported inside the request loop above and this iteration could be avoided.

Reply:
That loop is removed now, so only one iteration